src/data_gradients/managers/abstract_manager.py

```diff
@@ -126,36 +119,34 @@ def execute(self):
         )
 
         datasets_tqdm = tqdm(
-            zip_longest(self.train_iter, self.val_iter, fillvalue=None),
+            zip_longest(self.train_data.samples_iterator(split_name="train"), self.val_data.samples_iterator(split_name="val"), fillvalue=None),
             desc="Analyzing... ",
             total=self.n_batches,
         )
 
         self._train_iters_done, self._val_iters_done = 0, 0
         self._stopped_early = False
 
-        for i, (train_batch, val_batch) in enumerate(datasets_tqdm):
+        for i, (train_sample, val_sample) in enumerate(datasets_tqdm):
```

Comment:
is it a sample or a batch here?

Reply:
The `DatasetAdapter.__iter__` is the "default" way of using DatasetAdapter, which should return a batch of (`images`, `targets`) each of type numpy. 

The thing is that DG works by samples (each sample is an instance of a dataclass).
We used to build the samples with this `for sample in self.batch_processor.process(train_batch, split="train"):`.
So the processor returned samples. Now that the DatasetAdapter wraps the dataset AND the processor, there must be a way to return samples objects.

This is done with the method `samples_iterator ` (E.g. `self.train_data.samples_iterator(split_name="train"`).